edi_oca/models/edi_exchange_record.py

```diff
@@ -209,7 +211,13 @@ def record(self):
             return None
         if not self.model and self.parent_id:
             return self.parent_id.record
-        return self.env[self.model].browse(self.res_id).exists()
+        rec = self.env[self.model].browse(self.res_id).exists()
+        if rec and "company_id" in rec:
+            rec_company = rec.sudo().company_id
+            if rec_company and rec_company not in self.env.companies:
+                # Return empty recordset if company does not match
+                rec = rec[:0]
```

Comment:
```suggestion
                rec = self.env[self.model].browse()
```
IMO it's cleaner to show you're defining `rec` with an empty recordset.